lib/parser/parser-utils.ts

```diff
@@ -5,8 +5,55 @@
 import * as path from 'path';
 import * as htmlparser from 'htmlparser2';
 const merge = require('lodash/merge');
-import { DirectiveType, ComponentSet, ComponentsDefinition, Component, ParsedComponent, ComponentProperty } from '../models';
-import { GetFileContentType } from '..';
+import {
+    DirectiveType, ComponentSet, ComponentsDefinition, Component, ParsedComponent, ComponentProperty,
+    ComponentRendition, GetFileContentType
+} from '../models';
+
+/**
+ * Parses the components definition into the object which contains all needed data to make needed validations
+ * It already validates:
+ * - existing of component properties
+ * - existing of directive which properties point to
+ * - if directive keys are unique within a component html template
+ * - if group names are unique
+ * - if components have HTML rendition
+ *
+ * @param componentsDefinition
+ * @param getFileContent
+ * @returns Promise<ParsedComponentsDefinition>
+ */
+export async function parseDefinition(
+    componentsDefinition: ComponentsDefinition,
+    getFileContent?: GetFileContentType,
```

Comment:
Does it need to be optional? Could we make it required?

Reply:
But the componentsDefinition may contain the renditions, then getFileContent is not needed. Or what do you mean? Passing null or empty function?

Reply:
Never mind, leave it for now I guess. The design of the components set api in the backend makes it inconsistent.